src/main/java/com/xebialabs/deployit/ci/RepositoryUtils.java

```diff
@@ -20,22 +21,43 @@
 
```

Comment:
Can't we just overload getDeployitServer(String credentialName, String overridingCredentialId) having 
StandardUsernamePasswordCredentials overridingCredential = null;
if(null != overridingCredentialId) {
           overridingCredential = Credential.lookupSystemCredentials(overridingCredentialId);
           if(null == overridingCredential) {
                throw new IllegalArgumentException(Messages.DeployitNotifier_credentialIdNotFoundError(credentialId));
            }
}
return getDeployitServer(credentialName, overridingCredential)

Reply:
Cannot substitute Credentials instance with StandardUsernameCredentials instance.